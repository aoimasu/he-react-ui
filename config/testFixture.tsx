import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';

type Fixture = {
  component: React.ComponentType<any>;
  name?: string;
};

function testOneFixture(fixture: Fixture, variant: string = 'a') {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component as any;
  mount();
  test(`<${displayName || name} /> rendered correctly with ${fixture.name ||
    variant} fixture`, () => expect(getWrapper()).toMatchSnapshot());
}

// TODO: use real type guard functions to discriminate one fixture vs an array of them

export default function testAllFixtures(fixtures: any, variant?: string) {
  if (!('forEach' in fixtures)) {
    testOneFixture(fixtures as Fixture, variant);
  } else if (typeof fixtures.forEach === 'function') {
    (fixtures as Fixture[]).forEach(it => testOneFixture(it));
  }
}
