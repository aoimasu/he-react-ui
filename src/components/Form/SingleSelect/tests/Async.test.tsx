import fixture from '../fixtures/Async.fixture';
import testFixture from '../../../../../config/testFixture';

testFixture(fixture, 'Async');
