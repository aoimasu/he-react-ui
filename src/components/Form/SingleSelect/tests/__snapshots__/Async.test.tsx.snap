// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleSelect /> rendered correctly with Async fixture 1`] = `
<SingleSelect
  disabled={false}
  error=""
  id="demo"
  inline={false}
  loadOptionsAsync={
    [MockFunction] {
      "calls": Array [
        Array [
          "",
          [Function],
        ],
      ],
    }
  }
  name="demo"
  onBeforeOpen={[Function]}
  onChange={[Function]}
  onClose={[Function]}
  value={null}
>
  <div
    className="outer"
  >
    <div
      className="relative"
    >
      <div
        className="container"
      >
        <Async
          autoload={true}
          cache={false}
          className="select"
          disabled={false}
          id="demo"
          ignoreAccents={true}
          ignoreCase={true}
          joinValues={true}
          loadOptions={
            [MockFunction] {
              "calls": Array [
                Array [
                  "",
                  [Function],
                ],
              ],
            }
          }
          loadingPlaceholder="Loading..."
          name="demo"
          onChange={[Function]}
          onClose={[Function]}
          onOpen={[Function]}
          options={Array []}
          searchPromptText="Type to search"
          valueRenderer={[Function]}
        >
          <Select
            arrowRenderer={[Function]}
            autoload={true}
            autosize={true}
            backspaceRemoves={true}
            backspaceToRemoveMessage="Press backspace to remove {label}"
            cache={false}
            className="select"
            clearAllText="Clear all"
            clearRenderer={[Function]}
            clearValueText="Clear value"
            clearable={true}
            closeOnSelect={true}
            deleteRemoves={true}
            delimiter=","
            disabled={false}
            escapeClearsValue={true}
            filterOptions={[Function]}
            id="demo"
            ignoreAccents={true}
            ignoreCase={true}
            inputProps={Object {}}
            isLoading={false}
            joinValues={true}
            labelKey="label"
            loadOptions={
              [MockFunction] {
                "calls": Array [
                  Array [
                    "",
                    [Function],
                  ],
                ],
              }
            }
            loadingPlaceholder="Loading..."
            matchPos="any"
            matchProp="any"
            menuBuffer={0}
            menuRenderer={[Function]}
            multi={false}
            name="demo"
            noResultsText="Type to search"
            onBlurResetsInput={true}
            onChange={[Function]}
            onClose={[Function]}
            onCloseResetsInput={true}
            onInputChange={[Function]}
            onOpen={[Function]}
            onSelectResetsInput={true}
            openOnClick={true}
            optionComponent={[Function]}
            options={Array []}
            pageSize={5}
            placeholder="Select..."
            removeSelected={true}
            required={false}
            rtl={false}
            scrollMenuIntoView={true}
            searchPromptText="Type to search"
            searchable={true}
            simpleValue={false}
            tabSelectsValue={true}
            trimFilter={true}
            valueComponent={[Function]}
            valueKey="value"
            valueRenderer={[Function]}
          >
            <div
              className="Select select is-clearable is-searchable Select--single"
            >
              <input
                disabled={false}
                name="demo"
                type="hidden"
                value=""
              />
              <div
                className="Select-control"
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
              >
                <span
                  className="Select-multi-value-wrapper"
                  id="react-select-2--value"
                >
                  <div
                    className="Select-placeholder"
                  >
                    Select...
                  </div>
                  <AutosizeInput
                    aria-activedescendant="react-select-2--value"
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-owns=""
                    className="Select-input"
                    id="demo"
                    injectStyles={true}
                    minWidth="5"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    value=""
                  >
                    <div
                      className="Select-input"
                      style={
                        Object {
                          "display": "inline-block",
                        }
                      }
                    >
                      <input
                        aria-activedescendant="react-select-2--value"
                        aria-expanded="false"
                        aria-haspopup="false"
                        aria-owns=""
                        id="demo"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        required={false}
                        role="combobox"
                        style={
                          Object {
                            "boxSizing": "content-box",
                            "width": "5px",
                          }
                        }
                        value=""
                      />
                      <div
                        style={
                          Object {
                            "height": 0,
                            "left": 0,
                            "overflow": "scroll",
                            "position": "absolute",
                            "top": 0,
                            "visibility": "hidden",
                            "whiteSpace": "pre",
                          }
                        }
                      />
                    </div>
                  </AutosizeInput>
                </span>
                <span
                  className="Select-arrow-zone"
                  onMouseDown={[Function]}
                >
                  <span
                    className="Select-arrow"
                    onMouseDown={[Function]}
                  />
                </span>
              </div>
            </div>
          </Select>
        </Async>
        <DropDown
          className="caret"
          fill="currentColor"
          height={32}
          width={32}
        >
          <svg
            className="caret"
            fill="currentColor"
            height={32}
            viewBox="0 0 24 24"
            width={32}
          >
            <path
              d="M8.068 9h7.864a.5.5 0 0 1 .385.82l-3.933 4.72a.5.5 0 0 1-.768 0L7.683 9.82A.5.5 0 0 1 8.068 9z"
            />
          </svg>
        </DropDown>
      </div>
    </div>
  </div>
</SingleSelect>
`;
